### ISSUE TITLE
1.7.0 doesn't show the user went offline, only online

### ISSUE THREAD

**Issue by yurivict** [qTox/qTox#4010]
Previous versions did show "offline" records, and it makes sense, so this is a regression:

![qtox-no-offline-records](https://cloud.githubusercontent.com/assets/271906/21472904/a4061d00-caa3-11e6-94af-f40a14c6b474.png)

**Comment by AliceGrey** [comment]
I'll take a look at this. ~~Most likely happened with merge #3983~~ or maybe sometime earlier? Any ideas @Diadlo

**Comment by zetok** [comment]
@AliceGrey generally when in doubt, `git bisect` :-)

#3983 was merged after v1.7.0 was released, so that's not it.

https://github.com/qTox/qTox/wiki/Testing#regression-testing ← a bit on using bisect

f5011cdb2feb2da042e373b4e4094e967bde8120 is the last commit that works with the 0.0.0 irungentoo toxcore, and most likely already has the regression included.

Working version for @yurivict is most likely v1.5.2, perhaps even v1.6.0.

```
git bisect start
git bisect good v1.5.2
git bisect bad f5011cdb2feb2da042e373b4e4094e967bde8120
```

Note that testing most likely should be done on a separate profile, since the kind of version switching that `bisect` does would result in an ~erased history, settings, etc.

**Comment by AliceGrey** [comment]
Thank you @zetok. I'll take a look at that

**Comment by shinodem** [comment]
i just update to 1.7 and can't see friends in online at old versions of qtox

**Comment by yurivict** [comment]
In 1.7.1 friends are online ok for me.

**Comment by shinodem** [comment]
@yurivict Where i can find that build? Thanks in advance

**Comment by yurivict** [comment]
This depends on your system.

**Comment by zetok** [comment]
@AliceGrey any update?

**Comment by shinodem** [comment]
@yurivict win10 x64

**Comment by yurivict** [comment]
1.7.0 is supposed to be the same as 1.7.1 though. It might be a Windows problem.
I don't use Windows.

**Comment by shinodem** [comment]
it seams what 1.7.0 need much time sometimes 10 minutes sometimes 1 hour to connect to friends to see them in online.

**Comment by yurivict** [comment]
@shinodem You need to create a separate bug for this. This must be the windows problem, because I don't see this.

**Comment by zetok** [comment]
>This must be the windows problem, because I don't see this.

Not a windows problem, but a separate issue should be created.

**Comment by Diadlo** [comment]
Should be fixed in current master. Can anyone confirm?
But in current master there is status message duplication (will be fixed in #4066)

**Comment by yurivict** [comment]
Still the same problem with 3b56a6e

**Comment by Diadlo** [comment]
I can't reproduce

**Comment by yurivict** [comment]
It prints "online" message many times in a row. They should be strictly alternating.

**Comment by Diadlo** [comment]
@yurivict Friend just goes offline -> online -> offline -> online?
![spectacle ns7124](https://cloud.githubusercontent.com/assets/6306608/21799079/2baf68e4-d731-11e6-8c5c-532e9d83bf47.png)

**Comment by yurivict** [comment]
yes, same kind of message several times in a row,

**Comment by eagle500** [comment]
It happens to me also.

**Comment by zetok** [comment]
Still not fixed on 5a043599017472c16aa08265edfed9123c94
![spectacle wy1802](https://cloud.githubusercontent.com/assets/3148759/22624881/59a905e4-eb80-11e6-9d0e-842e34bb13ee.png)

NB: if someone's status isn't `Online`, messages differ:
![spectacle ax1802](https://cloud.githubusercontent.com/assets/3148759/22624887/7c945c7a-eb80-11e6-9ea4-18af5de77c60.png)
